### PR TITLE
chore: add scaffold-ui tests

### PR DESCRIPTION
**Pull request by tomiir**
# Description

- Adds tests for `scaffold-ui` utilities

## Type of change

- [X] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [X] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [X] My changes generate no new warnings
- [X] I have reviewed my own code
- [X] I have filled out all required sections
- [X] I have tested my changes on the preview link
- [] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/scaffold-ui/src/utils/WalletUtil.ts [hunk 1]
```diff
@@ -50,5 +50,30 @@ export const WalletUtil = {
...
+      .reduce<Record<string, boolean>>((acum, val) => {
+        if (!val.info?.rdns) {
+          return acum
+        }
+        acum[val.info.rdns] = true
+
+        return acum
+      }, {})
+
+    const walletsWithInstalled: (WcWallet & { installed: boolean })[] = wallets.map(wallet => ({
```
Comment (magiziz) [inline packages/scaffold-ui/src/utils/WalletUtil.ts:68]: Can we remove `(WcWallet & { installed: boolean })[]` part ?

File: packages/scaffold-ui/test/WalletUtil.test.ts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+}
+
+// Wallets
+const mockMetamaskWallet: WcWallet = { id: '1', name: 'Wallet 1', rdns: 'io.metamask' }
+const mockRainbowWallet: WcWallet = { id: '2', name: 'Wallet 2', rdns: 'io.rainbow' }
+const mockTrustWallet: WcWallet = { id: '3', name: 'Wallet 3', rdns: 'io.trustwallet' }
+const mockCoinbaseWallet: WcWallet = { id: '5', name: 'Wallet 5', rdns: 'io.coinbase' }
+
+describe('WalletUtil', () => {
+  const mockWallets: WcWallet[] = [mockMetamaskWallet, mockRainbowWallet, mockTrustWallet]
```
Comment (magiziz) [inline packages/scaffold-ui/test/WalletUtil.test.ts:49]: 👌

**Review by magiziz — APPROVED** [review]
lgtm

File: packages/scaffold-ui/test/WalletUtil.test.ts [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  describe('filterOutDuplicatesByRDNS', () => {
+    it('should filter out wallets with RDNS from connectors and recent wallets', () => {
+      const mockConnectors = [mockMetamaskConnector, mockRainbowConnector, mockCoinbaseconnector]
+      const mockRecentWallets = [mockTrustWallet]
+
+      vi.spyOn(ConnectorController.state, 'connectors', 'get').mockReturnValue(mockConnectors)
+      vi.spyOn(StorageUtil, 'getRecentWallets').mockReturnValue(mockRecentWallets)
+      vi.spyOn(CoreHelperUtil, 'isMobile').mockReturnValue(false)
+
+      console.log('>> ConnectorController.connectors', ConnectorController.state.connectors)
```
Comment (enesozturk) [inline packages/scaffold-ui/test/WalletUtil.test.ts:66]: Log here and below

**Pull request merged**